### PR TITLE
Add new instrumentation: hapi

### PR DESCRIPTION
**Pull request by unindented**
Fixes https://github.com/openzipkin/zipkin-js/issues/43

### CODE REVIEW

File: packages/zipkin-instrumentation-hapi/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const ctxImpl = new ExplicitContext();
+const recorder = new ConsoleRecorder();
+
+const tracer = new Tracer({ctxImpl, recorder}); // configure your tracer properly here
+
+const server = new Hapi.Server();
+
+// Add the Zipkin middleware
+server.register({
+  register: middleware,
```
Comment (eirslett) [inline packages/zipkin-instrumentation-hapi/README.md]: do you miss a `const middleware = ...` here somewhere?

File: packages/zipkin-instrumentation-hapi/test/integrationTest.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
+const sinon = require('sinon');
+const {Tracer, ExplicitContext} = require('zipkin');
+const Hapi = require('hapi');
+const middleware = require('../src/hapiMiddleware');
+
+describe('hapi middleware - integration test', () => {
+  it.skip('should create traceId', () => {});
```
Comment (eirslett) [inline packages/zipkin-instrumentation-hapi/test/integrationTest.js]: This one is probably not meant to be included?

**Comment by eirslett** [comment]
Thanks a lot!

File: packages/zipkin-instrumentation-hapi/README.md (hunk 1 above)
Comment (unindented) [inline packages/zipkin-instrumentation-hapi/README.md]: Ah, it should be `zipkinMiddleware`. I'll change that right meow.

File: packages/zipkin-instrumentation-hapi/test/integrationTest.js (hunk 2 above)
Comment (unindented) [inline packages/zipkin-instrumentation-hapi/test/integrationTest.js]: I copied the tests from `zipkin-instrumentation-express`, which included this pending one. Should I remove it?

Comment (unindented) [inline packages/zipkin-instrumentation-hapi/test/integrationTest.js]: @eirslett Let me know if you want it removed and I'll do it. Really looking forward to using your awesome library on my services!

Comment (eirslett) [inline packages/zipkin-instrumentation-hapi/test/integrationTest.js]: Yes, it would be nice to remove it.

Comment (unindented) [inline packages/zipkin-instrumentation-hapi/test/integrationTest.js]: Done!

Comment (unindented) [inline packages/zipkin-instrumentation-hapi/test/integrationTest.js]: @eirslett Done!

**Comment by adriancole** [comment]
Nice work folks

**Comment by unindented** [comment]
It looks like one of the tests timed out when merging into master... Maybe re-running will fix it?

One more thing: What's the release schedule for this project? When would the corresponding npm package get published?

**Comment by eirslett** [comment]
Usually @adriancole releases for every PR, or so. As long as the tests pass.

**Comment by adriancole** [comment]
Most effective is the explicit release request :)

**Comment by adriancole** [comment]
0.2.7 fresh off the press https://twitter.com/zipkinproject/status/783274339802112001

**Comment by ApsOps** [comment]
Please update the README as well :)

**Comment by adriancole** [comment]
>
> Please update the README as well :)
>
I double-dare you to do it!

**Comment by ApsOps** [comment]
Haha.. I wanted to do it but I was not sure what to write 🙈

**Comment by adriancole** [comment]
here's for adding hapi.js
https://github.com/openzipkin/zipkin-js/commit/107443de7d42f5e2581d89bd2aa821cad1c3b378

here's one on how to nag for a release :)
https://github.com/openzipkin/zipkin-js/commit/1e91435fd7aa9f399e921b699b83d7f0c1cdfad1

**Pull request merged**